include/math_utils.h

```diff
@@ -205,6 +205,16 @@ constexpr uint32_t clamp_to_uint32(const T val)
 	return static_cast<uint32_t>(std::clamp(val, min_val, max_val));
 }
 
+constexpr uint8_t read_low_nibble(const uint8_t byte)
+{
+	return static_cast<uint8_t>(byte % 0x10);
```

Comment:
`byte && 0x0f` and `(byte && 0xf0) >> 8`?
That's what I expect to see when working with nibbles, this gave me a pause.

Reply:
For higher nibble it is `byte >> 4`, actually. OK, changed - due to past experiences with mathematics I'm more used to division and remainder operators, but I admin I'm an exception :)

Reply:
Ah yeah, `>> 4` 😄  Well, I'm used to seeing boolean ops and bit shifts everywhere in low-level code from my asm coding days, and while it does not matter much these days performance-wise, division/modulo was very heavy compared to bit-ops back in the day. So yeah, partly tradition, partly familiarity for me 😄 But maybe performance still matters, division is never really "lightweight" on any platform.